streaming/src/test/scala/org/apache/spark/streaming/receiver/RateLimiterSuite.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.streaming.receiver
+
+import org.apache.spark.SparkConf
+import org.apache.spark.streaming.TestSuiteBase
+
+/** Testsuite for testing the network receiver behavior */
+class RateLimiterSuite extends TestSuiteBase {
+
+  test("rate limiter initializes even without a maxRate set") {
```

Comment:
These unit tests are insufficient. I expected unit tests in RateLimiterSuite to test whether setting the rate actually change the limit or not (that is, by pushing data through the rate limiter). It should be
- Create RateLimiter with a rate, push data and set whether the limit is satisfied
- Then increase the rate, and push data to see whether new rate is satisfied
- Then decrease the rate, and push data to see whether new rate is satisfied.